tests/runtime/ray/test_build_workflow.py

```diff
@@ -292,22 +292,23 @@ def test_handles_taskwrappingerror(
 
         # Given
         workflow = workflow_parametrised_with_resources().model
+        assert workflow.metadata is not None
         workflow.metadata.python_version = python_version
         client = create_autospec(_build_workflow.RayClient)
         client.add_options = Mock(side_effect=exceptions.TaskWrappingError)
 
         wf_run_id = "<wf run id sentinel>"
 
         # When
-        with pytest.raises(expected_exception):
+        with pytest.raises(expected_exception):  # type: ignore
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏The reason you're seeing a type error here is because `expected_exception` is typed as an `Exception` (which is an instance of an exception) but what you actually pass is `Type[Exception]`.

I'd remove `# type: ignore` and change `expected_exception: Exception` to `expected_exception: Type[Exception]`. (Type is from `typing`)